Library/Homebrew/requirements/blas_requirement.rb

```diff
@@ -0,0 +1,47 @@
+require "requirement"
+
+class BlasRequirement < Requirement
+  fatal true
+  default_formula "openblas"
+
+  # This ensures that HOMEBREW_BLAS_CFLAGS and HOMEBREW_BLAS_LDFLAGS
+  # are always set. It does _not_ add them to CFLAGS or LDFLAGS; that
+  # should happen inside the formula.
```

Comment:
you asked about documentation, not sure if that's what you mean, but perhaps one could quickly summarise the logic here, i.e. : to check the presence of system's blas we compile a test program with HOMEBREW_BLAS_CFLAGS and HOMEBREW_BLAS_LDFLAGS if they are set or use the default values "" and "-lblas -llapack". If compilation fails we fall back to openblas and set HOMEBREW_BLAS_CFLAGS and HOMEBREW_BLAS_LDFLAGS accordingly. Otherwise, user specified or default values are stored.